core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -2107,7 +2107,7 @@ class ReplicaManager(val config: KafkaConfig,
    * @param delta           The delta to apply.
    * @param newImage        The new metadata image.
    */
-  def applyDelta(delta: TopicsDelta, newImage: MetadataImage): Unit = {
+  def applyDelta(delta: TopicsDelta, newImage: MetadataImage, metadataVersion: MetadataVersionDelta): Unit = {
```

Comment:
This is here to illustrate one way we can inform broker components of a metadata.version change. With this approach, version changes can be handled in-band with other metadata changes.

Another way is to register a listener with MetadataVersionManager which would be out-of-band from other metadata updates.